File: CiaiControllerSDK/Models/FunctionData.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace CiaiControllerSDK.Models
{
    /// <summary>
    /// 功能接口数据
    /// </summary>
    public class FunctionData
    {
        /// <summary>
        /// 功能名称
        /// </summary>
        [JsonPropertyName("functionName")]
        public string FunctionName { get; set; }

        /// <summary>
        /// 指令ID
        /// </summary>
        [JsonPropertyName("instructionId")]
        public string InstructionId { get; set; }

        /// <summary>
        /// 耗材信息
        /// </summary>
        [JsonPropertyName("labwareInfo")]
        public Labware LabwareInfo { get; set; }

        /// <summary>
        /// 设备名称
        /// </summary>
        [JsonPropertyName("equipmentName")]
        public string EquipmentName { get; set; }

        /// <summary>
        /// 位置ID
        /// </summary>
        [JsonPropertyName("nestId")]
        public string NestId { get; set; }

        /// <summary>
        /// 用户ID
        /// </summary>
        [JsonPropertyName("userId")]
        public string UserId { get; set; }

        /// <summary>
        /// 任务ID
        /// </summary>
        [JsonPropertyName("taskId")]
        public string TaskId { get; set; }

        /// <summary>
        /// 功能参数
        /// </summary>
        [JsonPropertyName("functionParam")]
        public object FunctionParam { get; set; }
    }

    /// <summary>
    /// 耗材信息
    /// </summary>
    public class Labware
    {
        /// <summary>
        /// 耗材名称
        /// </summary>
        [JsonPropertyName("LabwareName")]
        public string LabwareName { get; set; }

        /// <summary>
        /// 容量
        /// </summary>
        [JsonPropertyName("capacity")]
        public string Capacity { get; set; }

        /// <summary>
        /// 行数
        /// </summary>
        [JsonPropertyName("capacityRow")]
        public int CapacityRow { get; set; }

        /// <summary>
        /// 列数
        /// </summary>
        [JsonPropertyName("capacityColumn")]
        public int CapacityColumn { get; set; }
    }

    /// <summary>
    /// 完成回调数据
    /// </summary>
    public class Finish
    {
        /// <summary>
        /// 完成状态: finish, error
        /// </summary>
        [JsonPropertyName("completion")]
        public string Completion { get; set; }

        /// <summary>
        /// 错误信息
        /// </summary>
        [JsonPropertyName("errorMsg")]
        public string ErrorMsg { get; set; }

        /// <summary>
        /// 指令ID
        /// </summary>
        [JsonPropertyName("instructionId")]
        public string InstructionId { get; set; }

        /// <summary>
        /// 位置ID
        /// </summary>
        [JsonPropertyName("nestId")]
        public string NestId { get; set; }

        /// <summary>
        /// 结果输出
        /// </summary>
        [JsonPropertyName("resultOutput")]
        public List<ResultOutput> ResultOutput { get; set; }

        public static Finish Success()
        {
            return new Finish { Completion = "finish" };
        }

        public static Finish Error(string errorMessage)
        {
            return new Finish
            {
                Completion = "error",
                ErrorMsg = errorMessage
            };
        }
    }

    /// <summary>
    /// 结果输出
    /// </summary>
    public class ResultOutput
    {
        /// <summary>
        /// 名称
        /// </summary>
        [JsonPropertyName("name")]
        public string Name { get; set; }

        /// <summary>
        /// 结果数据
        /// </summary>
        [JsonPropertyName("resultData")]
        public object ResultData { get; set; }
    }
}

File: CiaiControllerSDKForJava/src/main/java/com/ciai/controller/sdk/model/FunctionData.java
package com.ciai.controller.sdk.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * 功能接口数据
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
public class FunctionData {

    @JsonProperty("functionName")
    private String functionName;

    @JsonProperty("instructionId")
    private String instructionId;

    @JsonProperty("labwareInfo")
    private Labware labwareInfo;

    @JsonProperty("equipmentName")
    private String equipmentName;

    @JsonProperty("nestId")
    private String nestId;

    @JsonProperty("userId")
    private String userId;

    @JsonProperty("taskId")
    private String taskId;

    @JsonProperty("functionParam")
    private Object functionParam;

    // Getters and Setters
    public String getFunctionName() {
        return functionName;
    }

    public void setFunctionName(String functionName) {
        this.functionName = functionName;
    }

    public String getInstructionId() {
        return instructionId;
    }

    public void setInstructionId(String instructionId) {
        this.instructionId = instructionId;
    }

    public Labware getLabwareInfo() {
        return labwareInfo;
    }

    public void setLabwareInfo(Labware labwareInfo) {
        this.labwareInfo = labwareInfo;
    }

    public String getEquipmentName() {
        return equipmentName;
    }

    public void setEquipmentName(String equipmentName) {
        this.equipmentName = equipmentName;
    }

    public String getNestId() {
        return nestId;
    }

    public void setNestId(String nestId) {
        this.nestId = nestId;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public String getTaskId() {
        return taskId;
    }

    public void setTaskId(String taskId) {
        this.taskId = taskId;
    }

    public Object getFunctionParam() {
        return functionParam;
    }

    public void setFunctionParam(Object functionParam) {
        this.functionParam = functionParam;
    }

    /**
     * 耗材信息
     */
    @JsonInclude(JsonInclude.Include.NON_NULL)
    public static class Labware {
        @JsonProperty("LabwareName")
        private String labwareName;

        @JsonProperty("capacity")
        private String capacity;

        @JsonProperty("capacityRow")
        private int capacityRow;

        @JsonProperty("capacityColumn")
        private int capacityColumn;

        // Getters and Setters
        public String getLabwareName() {
            return labwareName;
        }

        public void setLabwareName(String labwareName) {
            this.labwareName = labwareName;
        }

        public String getCapacity() {
            return capacity;
        }

        public void setCapacity(String capacity) {
            this.capacity = capacity;
        }

        public int getCapacityRow() {
            return capacityRow;
        }

        public void setCapacityRow(int capacityRow) {
            this.capacityRow = capacityRow;
        }

        public int getCapacityColumn() {
            return capacityColumn;
        }

        public void setCapacityColumn(int capacityColumn) {
            this.capacityColumn = capacityColumn;
        }
    }
}
